src/Gui/DlgGeneralImp.cpp

```diff
@@ -179,6 +190,20 @@ void DlgGeneralImp::saveSettings()
     ui->RecentFiles->onSave();
     ui->EnableCursorBlinking->onSave();
     ui->SplashScreen->onSave();
+    ui->WBSLeftCorner->onSave();
+    ui->WBSRightCorner->onSave();
+    ui->WBSToolBar->onSave();
+    ui->WBSViewMenu->onSave();
+
+    //Reload workbench to take into account workbench selector settings by refreshing toolbars and menubar.
+    if (needReloadWb) {
+        Workbench* w = WorkbenchManager::instance()->active();
+        if (w) {
+            Application::Instance->activateWorkbench("NoneWorkbench");
+            Application::Instance->activateWorkbench(w->name().c_str());
+        }
+    }
+    needReloadWb = false;

```

Comment:
WTH is that ? This is super dirty way. Even harmful. What happens if someone disabled None WB ?
Moreover, `needReloadWb` may be unitialized. All this shall be removed.